soc/arm/microchip_mec/common/soc_gpio.c

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
```

Comment:
why is this not a drivers/gpio?

Reply:
I needed a way to read GPIO pin states for the I2C driver.  Each SoC has a table of pins vs I2C port numbers.  This is temporary until I can develop Device Tree tables for each chip mapping I2C port numbers/names to the pin nodes.

Reply:
what precludes using the gpio driver in the i2c driver?  not sure why it would need to be soc code?

Reply:
Honest answer is my lack of skill using Device Tree.
The HW allows any I2C controller to use any of the 16 I2C ports the HW may implement. I need a DT based table to look up the GPIO driver handle and pin bit position if run time port switching will be supported.  I suspect some customer will demand run time I2C port switching.